src/clp_ffi_py/ir/native/PyKeyValuePairLogEvent.cpp

```diff
@@ -0,0 +1,341 @@
+#include <clp_ffi_py/Python.hpp>  // Must always be included before any other header files
+
+#include "PyKeyValuePairLogEvent.hpp"
+
+#include <cstddef>
+#include <optional>
+#include <span>
+#include <type_traits>
+#include <utility>
+
+#include <clp/BufferReader.hpp>
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ffi/ir_stream/Deserializer.hpp>
+#include <clp/ffi/ir_stream/IrUnitType.hpp>
+#include <clp/ffi/ir_stream/Serializer.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/SchemaTree.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/time_types.hpp>
+#include <clp/type_utils.hpp>
+
+#include <clp_ffi_py/api_decoration.hpp>
+#include <clp_ffi_py/error_messages.hpp>
+#include <clp_ffi_py/ir/native/error_messages.hpp>
+#include <clp_ffi_py/Py_utils.hpp>
+#include <clp_ffi_py/PyObjectCast.hpp>
+#include <clp_ffi_py/PyObjectUtils.hpp>
+#include <clp_ffi_py/utils.hpp>
+
+using clp::ffi::ir_stream::IRErrorCode;
+
+namespace clp_ffi_py::ir::native {
+namespace {
+/**
+ * Class that implements `clp::ffi::ir_stream::IrUnitHandlerInterface` for deserializing log events.
+ */
+class IrUnitHandler {
+public:
+    // Methods that implement the `clp::ffi::ir_stream::IrUnitHandlerInterface` interface
+    [[nodiscard]] auto handle_log_event(clp::ffi::KeyValuePairLogEvent&& deserialized_log_event
+    ) -> IRErrorCode {
+        log_event.emplace(std::move(deserialized_log_event));
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    [[nodiscard]] static auto handle_utc_offset_change(
+            [[maybe_unused]] clp::UtcOffset utc_offset_old,
+            [[maybe_unused]] clp::UtcOffset utc_offset_new
+    ) -> IRErrorCode {
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    [[nodiscard]] static auto handle_schema_tree_node_insertion(
+            [[maybe_unused]] clp::ffi::SchemaTree::NodeLocator schema_tree_node_locator
+    ) -> IRErrorCode {
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    [[nodiscard]] static auto handle_end_of_stream() -> IRErrorCode {
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    // TODO: we should enable linting when clang-tidy config is up-to-date to allow simple classes.
```

Comment:
```suggestion
    // TODO: We should enable linting when clang-tidy config is up-to-date to allow simple classes.
```

This is a nit, but try to consistently use capitals after colons.